src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Query.cs

```diff
@@ -489,6 +494,14 @@ private async Task ExecuteInternal()
                 // Call the query execution callback
                 if (QueryCompleted != null)
                 {
+                    if (sqlConn != null)
+                    {
+                        // Update PID here as it may change upon restart and only updates upon query execution.
+                        var intPID = (sqlConn.GetUnderlyingConnection() as SqlConnection).ServerProcessId;
```

Comment:
We'll still possibly have a PID on a failed query, did you check to make sure the SPID is being updated in that case too?